cocos/editor-support/cocostudio/CCSGUIReader.cpp

```diff
@@ -1512,6 +1512,10 @@ Widget* WidgetPropertiesReader0300::widgetFromJsonDictionary(const rapidjson::Va
                         }
                         child->setPosition(Vec2(child->getPositionX() + widget->getAnchorPointInPoints().x, child->getPositionY() + widget->getAnchorPointInPoints().y));
                     }
+                    if (widget == NULL) {
```

Comment:
That is not necessarily true. In the code fragment: 

```
    for (int i = 0; i < elementCount; ++i) {
        std::string key = stChildArray[i].GetName(cocoLoader);
        std::string value = stChildArray[i].GetValue(cocoLoader);
        
        if (key == "classname" )
        {
            if (!value.empty())
            {
                classname = value;
                widget = this->createGUI(classname);
            }
            else
            {
                CCLOG("Warning!!! classname not found!");
            }
        }else if(key == "children"){
            childrenNode = &stChildArray[i];
        }else if(key == "options"){
            optionsNode = &stChildArray[i];
        }
```
There is the distinct possibility of widget not being assigned to because all keys are either equal to "children" or "options." From there, if reader is null, we will of course go to the following code:

```
else{
            CCLOG("Widget or WidgetReader doesn't exists!!!  Please check your csb file.");
        }
```

However, there is no assert statement here, meaning that we can continue on down to the following code fragment:

```
   //parse children
    if (nullptr != childrenNode) {
        rapidjson::Type tType22  = childrenNode->GetType(cocoLoader);
        if (tType22 == rapidjson::kArrayType) {
            
            int childrenCount = childrenNode->GetChildNum();
            stExpCocoNode* innerChildArray = childrenNode->GetChildArray(cocoLoader);
            for (int i=0; i < childrenCount; ++i) {
                rapidjson::Type tType  = innerChildArray[i].GetType(cocoLoader);
                
                if (tType == rapidjson::kObjectType) {
                    
                    Widget *child = widgetFromBinary(cocoLoader, &innerChildArray[i]);
                    
                    if (child)
                    {
                        PageView* pageView = dynamic_cast<PageView*>(widget);
                        if (pageView)
                        {
                            pageView->addPage(static_cast<Layout*>(child));
                        }
                        else
                        {
                            ListView* listView = dynamic_cast<ListView*>(widget);
                            if (listView)
                            {
                                listView->pushBackCustomItem(child);
                            }
                            else
                            {
                                if (dynamic_cast<Layout*>(widget))
                                {
                                    if (child->getPositionType() == ui::Widget::PositionType::PERCENT)
                                    {
                                        child->setPositionPercent(Vec2(child->getPositionPercent().x + widget->getAnchorPoint().x,
                                                                       child->getPositionPercent().y + widget->getAnchorPoint().y));
                                    }
                                    child->setPosition(Vec2(child->getPositionX() + widget->getAnchorPointInPoints().x,
                                                            child->getPositionY() + widget->getAnchorPointInPoints().y));
                                }
                                widget->addChild(child);
                            }
                        }
                    }
                    
                }
            }
        }
    }
```

In that case, then the widget will be null. Of course, that means then that I should have put the assert statement in the area I posted out above! Thanks for catching that; will you take care of that?